compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirOverrideChecker.kt

```diff
@@ -172,12 +173,30 @@ object FirOverrideChecker : FirClassChecker() {
         firTypeScope: FirTypeScope,
         context: CheckerContext,
     ) {
+        val overriddenMemberSymbols = firTypeScope.retrieveDirectOverriddenOf(member)
+
         if (!member.isOverride) {
+            if (overriddenMemberSymbols.isEmpty() ||
+                context.session.overridesBackwardCompatibilityHelper.overrideCanBeOmitted(overriddenMemberSymbols, context)
+            ) {
+                return
+            }
+            val kind = member.source?.kind
+            if (kind !is FirRealSourceElementKind && kind !is FirFakeSourceElementKind.PropertyFromParameter) return
+
+            val overridden = overriddenMemberSymbols.first().originalOrSelf()
```

Comment:
Not sure about the use of `first()` here. Is it because, as long as the overridden can _not_ be omitted, we will report `VIRTUAL_MEMBER_HIDDEN` _on_ the current overridden anyway, and thus picking anything is okay (to retrieve the containing class)? Is there any chance that the generated error message is not deterministic?

Reply:
It is relevant since we're putting the containing class in the message. But I would just match whatever FE 1.0 is doing here, whether it's first or last or whatever, as long as it's not too complicated. If it's complicated (such as sorting by containing class name) then just pick the first, doesn't matter too much. 🙂

Reply:
+1 to Mark's reply. BTW, this is deterministic since traversing type scope is deterministic (IIUC).